docs/proposals/scheduling/20240115-elastic-quota-support-job-level-oversold-management.md

```diff
@@ -0,0 +1,295 @@
+   128 worker and per worker need 8 gpu cards, meanwhile some job are small like need 1 worker and per worker need 1 gpu card.
+   The more small jobs, the more resource fragments to large jobs. Binding quota and nodes together will restrict the scope
+   of resource fragment within the quota, and users in one quota can communicate and coordinate to solve the fragment.
+
+Normally, cluster maintainer will select some nodes binding to quota, and non-preemptible pods can only run on these nodes.
+For preemptible pods, maintainer can choose some nodes belongs to self quota and other quotas, or even the whole cluster nodes
+to increase scheduling successful-probability. When non-preemptible pods need to schedule, it will preempt the preemptible
+pods those running on non-preemptible-quota-binding-nodes.
+
+We do not have specific requirements for the method of quota node binding. Cluster maintainer can use affinity\label to bind
+concrete nodes, or first patch related nodes with quota label, then fill pods with affinity, or use resourceFlavor to
+select some nodes for quota and fill pod needed inputs. For our practise, the cluster maintainer will use "patch related
+nodes with quota label, then fill pods with affinity" way, non-preemptible pods will only fill its own quota affinity,
+and preemptible pods will fill several quotas based on business workload needs.
+
+In the end, node and quota binding should not have affect to preemption, so the doc below only introduce the node and quota binding
+in user-view, the concrete preemption implementation has no connection with quota node binding concept.
+
+##### Job Level Preemption
+As mentioned above, in AI scenario, job has property of "AllOrNothing", which means the process of resource allocation
+and preemption must be executed by job level, not pod level. If we do preemption pod by pod as usual, and one of pod preemption
+failed, the previous pod successful preemption will be meaningless and harmful. There are other plugins, like "CoScheduling", 
+have job semantics, so we mark pods belong to a same job with the same label.
+
+The preemption strategy is as followed:
+1. Only preemptionPolicy=PreemptLowerPriority or `quota.scheduling.koordinator.sh/can-preempt=true` pods can trigger preempt, 
+   it depends on the plugin args, and only preemptible pods can be preempted.
+
+2. If non-preemptibleTotalUsed + non-preemptibleJobRequest <= minQuota, then non-preemptible job can trigger preempt.
+
+3. Because non-preemptible job can only run on its own quota binding nodes, so it can preempt all preemptible pods on these node
+   no matter the preemptible pods belong to same quotas or not(inner quota preempt + over quota preempt together), and no matter 
+   preemptible pods' quota used is smaller than min or not.
+
+4. If only part of pods of non-preemptible job can be assigned successfully in preemption dry-run process,
+   the preemption will not really happen.
```

Comment:
These are also repeated and it is recommended to reorganize them.

Reply:
done